### ISSUE TITLE
Aesthetics: Working hours (d) appears to be less than actual 

### ISSUE THREAD

**Issue by abelted** [microsoft/wpa#154]
Working hours (d) is defined in line 116 as end_hour - Start_hour minus 1. This appears to exclude the fact that the end_hour is different than the end time. 

Example:
If end_hour = 1700
End time = 1800

Comment also seems to indicate that this line is calculating the total workday span when it may be actually intended to calculate the official workday span with at least 1 one hour break. 

I believe the intention might be d <- (end_hour - start_hour)

 https://github.com/microsoft/wpa/blob/4b432bab6bef4e6637b56475604dddbbd15efd61/R/workpatterns_classify_bw.R#L116

**Comment by moralec** [comment]
Hi @abelted!

Thanks for raising this! I am gonna check the code again. I believe that line is intended to calculate the number of  hour-blocks that exist within the limits specified by the user, but I need to check the rest of the code to double check why we added a -1

**Comment by moralec** [comment]
Ok I think this is just sloppy code on our side. We defined as 1 hour less in line 116 and then in lines 201 - 210 we take this into consideration as part of the conditionals (so people with full standard hours are expected to have d+1 days). 

Probably worth cleaning the code a bit to make it more intuitive. 
@martinctc

**Comment by moralec** [comment]
I also tested with the demo environment and the classification for each user is correct. Will relabel this as an enhancement to have nicer code

**Comment by martinctc** [comment]
Hi @abelted and @moralec - thanks for the question and for looking into this. Here is an example illustrating the intent of the code: 

Example: official hour for work starts at **09:00** and stops at **18:00**

- `start_hour` threshold should be `"0900"`
- `end_hour` threshold should be `"1800"`
- `d` is calculated as 8, since it is currently calculated as `end_hour` minus `start_hour` minus 1. This is correct because the _slot times_ should be **09:00** to **17:00** - where **17:00** covers the interval between **17:00** and **18:00**. However, `d` should **not** be interpreted as the actual day span, which should have been `d + 1`. 
- In line 188, there is an actual variable for `Day_Span` which is simple deduction of `end_hour` from `start_hour`. 

To put it another way, between 9 AM and 6 PM, it is only mathematically possible for someone to be active for a maximum of 8 distinct hour blocks, and it cannot be 9. 

The classification should be correct, and I agree with @moralec that clarification is needed as to what the `d` variable represents so that it is not confused with `Day_Span`, as well as the interpretation of `end_hour` for the user. 

I'll have a look to see how this can be clarified both in the code and in the documentation, potentially refactoring the code so that the intent is clear for those trying to replicate the method.

**Comment by moralec** [comment]
Super clear. I think we can close this by improving the documentation of the function and the comments in the code :)
